File: tools/other/tools/fnv.py
import sys

from ..core.utilities import fnv
from ..core.utilities import run_project

if __name__ == "__main__":
  if len(sys.argv) < 2:
    print("Usage: {} <string>".format(sys.argv[0]))
    sys.exit(1)

  sys.exit(fnv(sys.argv[1]))
  

File: tools/other/fnv/fnv.cpp
#include "core/defines.hpp"
#include "core/formatters.hpp"

using namespace other;

void Usage() {
  println("FNV Hash Calculator");
  println("Usage: fnv <string>");
  println("   > Example: fnv hello");
}

int main(int argc, char* argv[]) {
  if (argc < 2) {
    Usage();
    return 1;
  }

  std::string arg = argv[1];
  uint64_t fnv = FNV(arg);
  println("FNV({}) = {}", arg, fnv);

  return 0;
}
